File: backend/app/domain/schemas/department.py
from pydantic import BaseModel

from .school import SchoolResponse


class DepartmentBase(BaseModel):
    name: str
    description: str
    coord_email: str
    secre_email: str
    school_id: int
    cost_center: int | None
    director: str | None

class DepartmentCreate(DepartmentBase):
    pass


class DeparmentUpdate(DepartmentBase):
    pass


class DeparmentInDB(DepartmentBase):
    id: int

    class Config:
        orm_mode = True


class DepartmentResponse(DeparmentInDB):
    school: SchoolResponse

File: client/src/app/core/interfaces/department.ts
import { SchoolResponse } from "./school";

export interface DepartmentBase {
    name: string;
    description: string;
    coord_email: string;
    secre_email: string;
    school_id: number;
    cost_center?: number;
}


export interface DepartmentInDB extends DepartmentBase {
    id: number;
}

export interface DepartmentResponse extends DepartmentInDB{
    school: SchoolResponse;
}
